Water Drink Water/src/clients/blazor.wa.tbd/Pages/Register.razor.cs

```diff
@@ -0,0 +1,45 @@
+﻿using blazor.wa.tbd.Services;
+using Microsoft.AspNetCore.Components;
+
+namespace blazor.wa.tbd.Pages;
+
+public partial class Register : ComponentBase
+{
+    private readonly RegistrationFormData formData = new();
+
+    [Inject] public NavigationManager NavigationManager { get; set; } = null!;
+    [Inject] private AuthService AuthService { get; set; } = null!;
+
+    public string ValidationMessage { get; set; } = string.Empty;
+
+    public async Task ExecuteRegistration()
+    {
+        if (formData.IsValid)
+        {
+            var result = await AuthService.Register(formData.Name, formData.Email, formData.Password);
+
+            if (result)
+            {
+                NavigationManager.NavigateTo("/login");
+            }
+
+            ValidationMessage = "Unable to register, please try again.";
+        }
+        else
+        {
+            ValidationMessage = "Please fill out all fields and ensure passwords match.";
+        }
+    }
+
+    public class RegistrationFormData
+    {
+        public string Name { get; set; }
+        public string Email { get; set; }
+        public string Password { get; set; }
+        public string ConfirmPassword { get; set; }
```

Comment:
```suggestion
       [Required]
        [StringLength(50, ErrorMessage = "The {0} must be at least {2} and at most {1} characters long.", MinimumLength = 2)]
        public string Name { get; set; }
        [Required]
        [EmailAddress]
        public string Email { get; set; }
        [Required]
        [StringLength(100, ErrorMessage = "The {0} must be at least {2} and at most {1} characters long.", MinumumLength = 6]
        public string Password { get; set; }
        [Required]
        [Compare("Password", ErrorMessage = "The password and confirmation password do not match.")]
        public string ConfirmPassword { get; set; }
```

This will give more concrete feedback to the user.

Reply:
Validation Attributes are useful when being used with Blazor here, but they aren't useful unless we use the same model and they aren't useful if another client is produced using a different technology. We'll have to discuss validation at a greater level, but I can definitely appreciate how this would help.